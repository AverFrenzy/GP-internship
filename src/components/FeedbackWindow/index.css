.white-fill {
    fill: white;
    cursor: pointer;
}

.yellow-fill {
    fill: yellow;
    cursor: pointer;
}

.feedback-form {
    margin: 50px auto;
    padding: 20px 20px;
    width: 550px;
    background-color: #14de9b;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    border-radius: 15px;
}

.delete-rating-button {
    display: flex;
    justify-content: center;
    align-items: center;
    position: absolute;
    top: 0;
    right: -60px;
    width: 40px;
    height: 40px;
    background-color: #f17151;
    border-radius: 8px;
    cursor: pointer;
}

.form-item {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-top: 30px;
    width: 95%;
}

.form-item:focus {
    border: 2px solid yellow;
}

.rating-button {
    background-color: #13e2e7;
    border-radius: 10px;
    width: 95%;
    margin: 20px auto;
    height: 55px;
    font-size: 24px;
    color: white;
    cursor: pointer;
}

.rating-label {
    align-self: flex-start;
    padding: 0 0 4px 8px;
    font-size: 20px;
    color: white;
}
.input-extra {
    height: 55px;
    font-size: 26px;
    padding-left: 16px;
    border-radius: 10px;
    margin-bottom: 20px;
    width: 100%;
}

.input-extra:focus {
    border: 3px solid yellow;
}

.add-input-btn {
    width: 40px;
    height: 40px;
    padding: 6px;
    background-color: green;
    border-radius: 8px;
    cursor: pointer;
    position: absolute;
    top: 0;
    left: -54px;
}
.name-wrapper {
    position: relative;
    display: flex;
}
.input-phone {
    height: 55px;
    font-size: 26px;
    padding-left: 16px;
    border-radius: 10px;
    width: 100%;
}

.input-phone:focus {
    border: 3px solid yellow;
}

.name-input {
    position: relative;
    font-size: 34px;
    color: white;
}


.error-message {
    margin-left: 6px;
    margin-top: 10px;
    font-size: 20px;
    font-weight: bold;
    color: #f17151;
}

.error-message:before {
    display: inline;
    content: "âš  ";
}

.rating-item {
    display: none;
}

.comment-text {
    width: 100%;
    height: 150px;
    font-size: 26px;
    padding: 8px 0 0 16px;
    border-radius: 10px;
}

.comment-text:focus {
    border: 3px solid yellow;
}