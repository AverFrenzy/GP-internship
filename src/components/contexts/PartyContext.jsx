import React, { createContext, useContext, useEffect, useState } from 'react';

import { getDietsInfo, getParticipantsInfo } from '../../services/pizzaService';


const PartyContext = createContext();

export const usePartyContext = () => useContext(PartyContext);

export const PartyContextProvider = ({ children }) => {
  const [partyInfo, setPartyInfo] = useState([]);
  const [extraFields, setExtraFields] = useState([]);
  const [isLoading, setIsLoading] = useState(false);

  const fetchData = async () => {
    setIsLoading(true);
    try {
      const participantsData = await getParticipantsInfo();
      const dietData = await getDietsInfo(participantsData.party.map(person => person.name));
      const vegansList = dietData.diet.filter(person => person.isVegan).map(person => person.name);
      const totalPartyInfo = participantsData.party.map((person, index) => {
        return {
          id: index + 1,
          name: person.name,
          eatsPizza: person.eatsPizza,
          isVegan: vegansList.includes(person.name),
          preRating: '3',
        };
      });
      setPartyInfo(totalPartyInfo);
      localStorage.setItem('totalPartyInfoStorage', JSON.stringify(totalPartyInfo));
    } catch (err) {
      console.log(err);
    } finally {
      setIsLoading(false);
    }
  };

  const deleteFeedback = (guest, reset, navigate) => {
    const newPartyInfo = [...partyInfo];
    newPartyInfo.forEach((person) => {
      if (person.name === guest.name) {
        person.isAddedFeedback = false;
        delete person.isAddedFeedback;
        delete person.phoneNumber;
        delete person.feedbackComment;
        delete person.feedbackRating;
      }
    });
    reset();
    setPartyInfo(newPartyInfo);
    localStorage.removeItem('totalPartyInfoStorage');
    localStorage.setItem('totalPartyInfoStorage', JSON.stringify(newPartyInfo));
    navigate('/');
  };

  const chooseRating = (event, guest) => {
    const newPartyInfo = [...partyInfo];
    newPartyInfo.forEach((person) => {
      if (person.name === guest.name) {
        person.preRating = event.target.value;
      }
    });
    setPartyInfo(newPartyInfo);
  };

  const addFeedBack = (guest, data) => {
    const newPartyInfo = [...partyInfo];
    newPartyInfo.forEach((person) => {
      if (person.name === guest.name) {
        person.isAddedFeedback = true;
        person.feedbackRating = data.rating;
        person.phoneNumber = data.phoneNumber;
        person.preRating = '3';
        person.feedbackComment = data.feedbackComment;
      }
    });
    setPartyInfo(newPartyInfo);
    localStorage.removeItem('totalPartyInfoStorage');
    localStorage.setItem('totalPartyInfoStorage', JSON.stringify(newPartyInfo));
  };

  const addInput = (event) => {
    event.stopPropagation()
    const duplicateFields = [<input className='input-extra' key={extraFields.length}/>, ...extraFields];
    console.log(duplicateFields);
    setExtraFields(duplicateFields);
  };

  const refreshList = () => {
    localStorage.removeItem('totalPartyInfoStorage');
    fetchData();
  };

  useEffect(() => {
    if (localStorage.hasOwnProperty('totalPartyInfoStorage')) {
      setPartyInfo(JSON.parse(localStorage.getItem('totalPartyInfoStorage')));
    } else {
      fetchData();
    }
  }, []);

  const value = {
    partyInfo,
    isLoading,
    addFeedBack,
    chooseRating,
    refreshList,
    deleteFeedback,
    extraFields,
    addInput
  };

  return (
    <PartyContext.Provider value={ value }>
      { children }
    </PartyContext.Provider>
  );
};
