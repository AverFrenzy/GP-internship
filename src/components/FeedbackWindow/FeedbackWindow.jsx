import React from 'react';
import { useParams, useNavigate } from 'react-router-dom';
import { useForm } from 'react-hook-form' ;

import { usePartyContext } from '../contexts/PartyContext';
import RatingStarSVG from '../svg/RatingStarSVG';
import DeleteRatingButtonSVG from '../svg/DeleteRatingButtonSVG';
import './index.css';
import PlusSVG from '../svg/PlusSVG';


export const FeedbackWindow = () => {
  const { id } = useParams();
  const navigate = useNavigate();
  const { partyInfo, addFeedBack, chooseRating, deleteFeedback, extraFields, addInput } = usePartyContext();
  const guest = partyInfo.find(guest => +id === guest.id);
  const {
    register,
    formState: {
      errors,
      isValid
    },
    setError,
    clearErrors,
    handleSubmit,
    reset,
    getValues,
    setValue,
  } = useForm({
    mode: 'onBlur',
    defaultValues: {
      phoneNumber: guest?.phoneNumber,
      rating: guest?.feedbackRating ? guest?.feedbackRating : '3',
      feedbackComment: guest?.feedbackComment
    }
  });

  const onSubmit = (data) => {
    addFeedBack(guest, data);
    navigate('/');
  };

  return (
    <form className='feedback-form' data-testid='feedback-form' onSubmit={ handleSubmit(onSubmit) }>
      <div className='name-wrapper'>
        <div className='add-input-btn' data-testid='add-input-btn' onClick={ addInput }>
          <PlusSVG />
        </div>
        <span className='name-input' data-testid='name-input'>{ ` ${ guest?.name }'s feedback ` }
          { guest?.isAddedFeedback &&
            <div onClick={ () => deleteFeedback(guest, reset, navigate) } className='delete-rating-button'>
              <DeleteRatingButtonSVG />
            </div> }
        </span>
      </div>
      <div className='form-item' data-testid='form-item'>
        { !!extraFields.length && extraFields }
        <label htmlFor='form-phone-label' className='rating-label'>Phone number</label>
        <input id='form-phone-label' className='input-phone' disabled={ guest?.isAddedFeedback } autoComplete='off'
               placeholder='Enter your phone number'
               { ...register('phoneNumber', {
                 required: 'Phone number is required',
                 minLength: {
                   value: 3,
                   message: 'Your phone number must have at least 3 chars'
                 },
                 maxLength: {
                   value: 10,
                   message: 'Your phone number can\'t contain more than 10 chars'
                 },
                 onChange: () => {
                   const re = /^[+0-9][- +()0-9]*$/;
                   if (!re.test(getValues('phoneNumber'))) {
                     setError('phoneNumber', {
                         type: 'invalid number',
                         message: 'Telephone number can\'t contain letters',
                       },
                       { shouldFocus: true }
                     );
                     setValue('phoneNumber', getValues('phoneNumber').slice(0, -1));
                   } else {
                     clearErrors('phoneNumber');
                   }
                 },
               }) }
        />
        { errors?.phoneNumber && <div className='error-message'>{ errors?.phoneNumber?.message }</div> }
      </div>
      <div className='form-item' data-testid='form-item'>
        <div className='rating-wrapper'>
          { [...Array(5)].map((star, index) => {
            return (
              <label key={ index + 'ratingLabel' }>
                <input key={ index + 'ratingInput' } type='radio' className='rating-item'
                       disabled={ guest?.isAddedFeedback }
                       value={ index + 1 } { ...register('rating', {
                  onChange: (event) => {
                    chooseRating(event, guest);
                  }
                }) } />
                <RatingStarSVG
                  starColor={ (guest?.feedbackRating || guest?.preRating) >= index + 1 ? 'yellow-fill' : 'white-fill' }
                  key={ index + 'ratingStar' } />
              </label>
            );
          }) }
        </div>
      </div>
      <div className='form-item' data-testid='form-item'>
        <label htmlFor='form-comment-label' className='rating-label'>Comment</label>
        <textarea id='form-comment-label' className='comment-text'
                  placeholder='Leave your comment'
                  disabled={ guest?.isAddedFeedback } { ...register('feedbackComment') } />
      </div>
      { !guest?.isAddedFeedback &&
        (isValid ? <input className='rating-button' type='submit' value='Submit' />
          : <button className='rating-button' data-testid='cancel-btn' onClick={ () => navigate('/') }>Cancel</button>) }
    </form>
  );
};
