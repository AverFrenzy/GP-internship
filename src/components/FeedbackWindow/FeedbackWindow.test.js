import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

import { FeedbackWindow } from './FeedbackWindow';
import { PartyContextProvider } from '../contexts/PartyContext';


describe('FeedbackWindow ', () => {
  it('render component', async () => {
    render(
      <MemoryRouter>
        <PartyContextProvider>
          <FeedbackWindow />
        </PartyContextProvider>
      </MemoryRouter>
    );
    const feedbackForm = await screen.getByTestId('feedback-form');
    const addInputBtn = await screen.getByTestId('add-input-btn');
    const nameInput = await screen.getByTestId('name-input');
    const formItems = await screen.getAllByTestId('form-item');
    const cancelBtn = await screen.getByTestId('cancel-btn');

    expect(feedbackForm).toBeInTheDocument();
    expect(addInputBtn).toBeInTheDocument();
    expect(nameInput).toBeInTheDocument();
    expect(cancelBtn).toBeInTheDocument();
    expect(formItems).toHaveLength(3);
  });
  it('FeedbackWindow snapshot', () => {
    const FeedbackWindowSnapshot = render(
      <MemoryRouter>
        <PartyContextProvider>
          <FeedbackWindow />
        </PartyContextProvider>
      </MemoryRouter>
    );
    expect(FeedbackWindowSnapshot).toMatchSnapshot();
  });
});
