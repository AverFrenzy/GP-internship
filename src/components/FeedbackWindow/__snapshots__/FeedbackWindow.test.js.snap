// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackWindow  FeedbackWindow snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="feedback-form"
        data-testid="feedback-form"
      >
        <div
          class="name-wrapper"
        >
          <div
            class="add-input-btn"
            data-testid="add-input-btn"
          >
            <svg
              class="hhh"
              id="Layer_1"
              style="enable-background: new 0 0 512 512;"
              version="1.1"
              viewBox="0 0 512 512"
              x="0px"
              xml:space="preserve"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
              y="0px"
            >
              <g>
                <path
                  d="M303.02,208.98V0H208.98v208.98H0v94.041h208.98V512h94.041V303.02H512V208.98H303.02z M480.653,271.673h-208.98v208.98
 h-31.347v-208.98H31.347v-31.347h208.98V31.347h31.347v208.98h208.98V271.673z"
                />
              </g>
            </svg>
          </div>
          <span
            class="name-input"
            data-testid="name-input"
          >
             undefined's feedback 
          </span>
        </div>
        <div
          class="form-item"
          data-testid="form-item"
        >
          <label
            class="rating-label"
            for="form-phone-label"
          >
            Phone number
          </label>
          <input
            autocomplete="off"
            class="input-phone"
            id="form-phone-label"
            name="phoneNumber"
            placeholder="Enter your phone number"
          />
        </div>
        <div
          class="form-item"
          data-testid="form-item"
        >
          <div
            class="rating-wrapper"
          >
            <label>
              <input
                class="rating-item"
                name="rating"
                type="radio"
                value="1"
              />
              <svg
                viewBox="0 0 576 512"
                width="75"
              >
                <path
                  class="white-fill"
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
                />
              </svg>
            </label>
            <label>
              <input
                class="rating-item"
                name="rating"
                type="radio"
                value="2"
              />
              <svg
                viewBox="0 0 576 512"
                width="75"
              >
                <path
                  class="white-fill"
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
                />
              </svg>
            </label>
            <label>
              <input
                class="rating-item"
                name="rating"
                type="radio"
                value="3"
              />
              <svg
                viewBox="0 0 576 512"
                width="75"
              >
                <path
                  class="white-fill"
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
                />
              </svg>
            </label>
            <label>
              <input
                class="rating-item"
                name="rating"
                type="radio"
                value="4"
              />
              <svg
                viewBox="0 0 576 512"
                width="75"
              >
                <path
                  class="white-fill"
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
                />
              </svg>
            </label>
            <label>
              <input
                class="rating-item"
                name="rating"
                type="radio"
                value="5"
              />
              <svg
                viewBox="0 0 576 512"
                width="75"
              >
                <path
                  class="white-fill"
                  d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
                />
              </svg>
            </label>
          </div>
        </div>
        <div
          class="form-item"
          data-testid="form-item"
        >
          <label
            class="rating-label"
            for="form-comment-label"
          >
            Comment
          </label>
          <textarea
            class="comment-text"
            id="form-comment-label"
            name="feedbackComment"
            placeholder="Leave your comment"
          />
        </div>
        <button
          class="rating-button"
          data-testid="cancel-btn"
        >
          Cancel
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="feedback-form"
      data-testid="feedback-form"
    >
      <div
        class="name-wrapper"
      >
        <div
          class="add-input-btn"
          data-testid="add-input-btn"
        >
          <svg
            class="hhh"
            id="Layer_1"
            style="enable-background: new 0 0 512 512;"
            version="1.1"
            viewBox="0 0 512 512"
            x="0px"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
            y="0px"
          >
            <g>
              <path
                d="M303.02,208.98V0H208.98v208.98H0v94.041h208.98V512h94.041V303.02H512V208.98H303.02z M480.653,271.673h-208.98v208.98
 h-31.347v-208.98H31.347v-31.347h208.98V31.347h31.347v208.98h208.98V271.673z"
              />
            </g>
          </svg>
        </div>
        <span
          class="name-input"
          data-testid="name-input"
        >
           undefined's feedback 
        </span>
      </div>
      <div
        class="form-item"
        data-testid="form-item"
      >
        <label
          class="rating-label"
          for="form-phone-label"
        >
          Phone number
        </label>
        <input
          autocomplete="off"
          class="input-phone"
          id="form-phone-label"
          name="phoneNumber"
          placeholder="Enter your phone number"
        />
      </div>
      <div
        class="form-item"
        data-testid="form-item"
      >
        <div
          class="rating-wrapper"
        >
          <label>
            <input
              class="rating-item"
              name="rating"
              type="radio"
              value="1"
            />
            <svg
              viewBox="0 0 576 512"
              width="75"
            >
              <path
                class="white-fill"
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
              />
            </svg>
          </label>
          <label>
            <input
              class="rating-item"
              name="rating"
              type="radio"
              value="2"
            />
            <svg
              viewBox="0 0 576 512"
              width="75"
            >
              <path
                class="white-fill"
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
              />
            </svg>
          </label>
          <label>
            <input
              class="rating-item"
              name="rating"
              type="radio"
              value="3"
            />
            <svg
              viewBox="0 0 576 512"
              width="75"
            >
              <path
                class="white-fill"
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
              />
            </svg>
          </label>
          <label>
            <input
              class="rating-item"
              name="rating"
              type="radio"
              value="4"
            />
            <svg
              viewBox="0 0 576 512"
              width="75"
            >
              <path
                class="white-fill"
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
              />
            </svg>
          </label>
          <label>
            <input
              class="rating-item"
              name="rating"
              type="radio"
              value="5"
            />
            <svg
              viewBox="0 0 576 512"
              width="75"
            >
              <path
                class="white-fill"
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-56.4 0z"
              />
            </svg>
          </label>
        </div>
      </div>
      <div
        class="form-item"
        data-testid="form-item"
      >
        <label
          class="rating-label"
          for="form-comment-label"
        >
          Comment
        </label>
        <textarea
          class="comment-text"
          id="form-comment-label"
          name="feedbackComment"
          placeholder="Leave your comment"
        />
      </div>
      <button
        class="rating-button"
        data-testid="cancel-btn"
      >
        Cancel
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
